### ISSUE TITLE
Incorrect role is defined for controls ‘Bold, italic, Align right’ etc. present under ‘Typography’ section

### ISSUE THREAD

**Issue by azaslonov** [paperbits/paperbits-demo#85]
User Experience:

User would find difficulty to identify the controls if role are not defined correctly for controls as they won't get the context of that control and also they won't be able to perform functionality  relevant to that control.

Note: User credentials should NOT be included in the bug.

 

Repro Steps:

1. Open Azure portal with valid credentials and create a API Management.

2. Click Developer Portal button to navigate to the portal.

3. Wait portal to load in design mode

4. Navigate to the Left nav and select "Styles" button.

5. Navigate to style page and select "Add variation" button.

6. Navigate to the various controls of the "Add variation" dialog box.

7. Open Developer tool using F12 key and  navigate to controls ‘Bold, italic, Align right’ etc. present under ‘Typography’ section.

8. Verify whether role is defined correct or not.

 

Actual Result:

Incorrect role is defined for controls ‘Bold, italic, Align right’ etc. present under ‘Typography’ section, its role is defined as “link” but  its behavior look like ‘button’, it will be confusing for screen reader users.

 

Expected Result:

Role should be define as ‘button' for the ‘Bold, italic, Align right’ etc. present under ‘Typography’ section, so that user can get correct information about control type.

 

Note:

This issue also exists with Chrome.

Similar issue exist throughout the application.

State also not defined for ‘Bold, italic, Align right’ etc. present under ‘Typography’ section.

 

MAS Reference:

MAS 4.1.2 - Name, Role, Value

 

Reference Links

Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information please use this Link. “HCL Staff should not log any third party external bugs. Should be routed to edad team”.
Please reach out to C&AI Teams channel for any process related queries.
 

 
![image](https://user-images.githubusercontent.com/2320302/76664959-587ca780-6543-11ea-947f-286fc0a20e0b.png)

**Comment by azaslonov** [comment]
Fixed.
